cli/schemas/config-file.json

```diff
@@ -0,0 +1,143 @@
+{
+  "$id": "https://deno.land/schemas/config-file.json",
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "description": "A JSON representation of a Deno configuration file.",
+  "required": [],
+  "title": "Deno configuration file Schema",
+  "type": "object",
+  "properties": {
+    "compilerOptions": {
```

Comment:
I think this is the syntax, but I'm not sure (something along these lines):

```json
"compilerOptions": {
  "$ref": "http://json.schemastore.org/tsconfig#/definitions/compilerOptionsDefinition/properties/compilerOptions"
}
```

That said, perhaps we should limit this to only the compiler options we support and I believe that would require copy and paste or at least using a "$ref" like this a bunch of times for each property we support.

Reply:
For now I went with copy-paste from http://json.schemastore.org/tsconfig based on the list here: https://deno.land/manual@v1.13.2/typescript/configuration#how-deno-uses-a-configuration-file it still needs updating defaults though (and `lib` property needs updating of list)

Reply:
yeah, I don't think we want to reference the external schema, as it doesn't meet our needs.